### DISCUSSION TITLE
additional set of buffers in consult-buffer-sources

### DISCUSSION THREAD

**Discussion by TitusMu** [minad/consult#960] [Q&A] [answered]
I would love to have the ephemeral buffers sorted in an extra category below the regular buffers when calling consult-buffer. The following definition doesn't work. It does sort them as wanted, but when I choose them, the buffer doesn't switch to *Messages* or *scratch*, it just stays as the current buffer. (The regular buffers can be chosen in the upper part of the consult-menu as usual.)

What did I do wrong?

```
    (setq consult-buffer-sources
          '(consult--source-hidden-buffer 
            consult--source-modified-buffer
            consult--source-buffer
            (:name "Ephemeral" :narrow 109
                 :category buffer :items
                   ("*Messages*"
                    "*scratch*"
                    "*Async-native-compile-log*"
                    "*Bookmark List*"))))

    (setq consult-buffer-filter
          '("\\` "
            "\\`\\*Completions\\*\\'"
            "\\`\\*scratch\\*\\'"
            "\\`\\*Messages\\*\\'"
            "\\`\\*Bookmark List\\*\\'"
            "\\`\\*EMMS-Log\\*\\'"
            "\\`\\*Flymake log\\*\\'"
            "\\`\\*Semantic SymRef\\*\\'"
            "\\`\\*tramp/.*\\*\\'"))

```
I need to filter them, so they are not listed twice, but just in the extra category. And neither :category buffer nor :category 'buffer works.

I am grateful for any advice.

I am using Emacs 30.0.50 and Consult 1.3.

**Comment by minad** [comment, marked as answer]
You missed the `:state` field, which defines preview and the action. Without the `:state` (or alternatively `:action`) field `consult--multi` does not know how to act on the selected candidate. See `consult--source-buffer` and `consult--source-recent-file` for comparison.

~~~ elisp
    (setq consult-buffer-sources
          '(consult--source-hidden-buffer 
            consult--source-modified-buffer
            consult--source-buffer
            (:name     "Ephemeral"
             :narrow   ?m
             :category buffer
             :state    consult--buffer-state
             :items
             ("*Messages*"
              "*scratch*"
              "*Async-native-compile-log*"
              "*Bookmark List*"))))
~~~

**Reply by TitusMu** [reply]
Thank you so much! This is the solution, it works wonderfully now.
